File: Euclid/Network/GameChannelInitializer.cs
using DotNetty.Transport.Channels;
using Euclid.Network.Codec;

namespace Euclid.Network
{
    internal class GameChannelInitializer : ChannelInitializer<IChannel>
    {
        protected override void InitChannel(IChannel channel)
        {
            IChannelPipeline pipeline = channel.Pipeline;
            pipeline.AddLast("gameEncoder", new NetworkEncoder());
            pipeline.AddLast("gameDecoder", new NetworkDecoder());
            pipeline.AddLast("clientHandler", new GameNetworkHandler());
        }
    }
}

File: euclid/src/main/java/euclid/network/GameChannelInitializer.java
package euclid.network;

import euclid.network.codec.NetworkDecoder;
import euclid.network.codec.NetworkEncoder;
import io.netty.channel.ChannelInitializer;
import io.netty.channel.socket.SocketChannel;

public class GameChannelInitializer extends ChannelInitializer<SocketChannel> {

    @Override
    protected void initChannel(SocketChannel channel) {
        channel.pipeline().addLast("gameEncoder", new NetworkEncoder());
        channel.pipeline().addLast("gameDecoder", new NetworkDecoder());
        channel.pipeline().addLast("clientHandler", new GameNetworkHandler());
    }
}
